app/controllers/users_controller.rb

```diff
@@ -0,0 +1,27 @@
+# require './config/environment'
+
+class UsersController < ApplicationController
+
+  get '/users/new' do
+    erb :"users/new.html"
+  end
+
+  post '/users' do
+
+      user = User.new(email: params[:email], password: params[:password], password_confirmation: params[:password_confirmation])
+      if user.save
+        flash.now[:notice] = "Thanks for signing up! You may now log in!"
+
+        redirect '/sessions/new'
+      else
+      # so what your typed for email can stay even if you get the password/confirmed passwd wrong  
+        @email = params[:email]
```

Comment:
Rather than storing the email they put in the form in an instance variable, you can pass the `user` object into the `erb` template via. `locals`